build.proj

```diff
@@ -116,6 +117,18 @@
     </DownloadFile>
   </Target>
 
+  <!-- Delete testhost.dll and testhost.exe from output folder,
+  start test from dotnet.exe to keep consistent behavior with older version of Microsoft.NET.Test.Sdk -->
+  <Target Name="DeleteTestHost">
+    <Message Importance="High" Text="Delete testhost.dll and testhost.exe from output folder..." />
```

Comment:
question: Why is this necessary?

Reply:
This also related to upgrade package "Microsoft.NET.Test.Sdk". Before upgrade the package, testhost.dll and testhost.exe are not copied to output path and tests are all starting from dotnet.exe. After upgrade package, tests will starts from testhost which will cause 2 issues:
1. RemoteExecutor will not able to start new process as below line will get testhost instead of dotnet: [https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.TestFramework/RemoteExecutor.cs#L29](https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.TestFramework/RemoteExecutor.cs#L29)
2. For x86 tests, we should use testhost.x86 instead of testhost directly, this should be achieved by set /p:Configuration instead of buildArch(which is ML.NET currently using), but setting Configuration will modify the build behavior a lot and corrupt the ML.NET build system.

So, based on these 2 reason, I'm add this target to remove testhost after build to keep consistent behavior as before upgrade test sdk package.

---
In reply to: [432262763](https://github.com/dotnet/machinelearning/pull/5169#discussion_r432262763) [](ancestors = 432262763)